src/macro/macroApi.ml

```diff
@@ -1477,7 +1477,7 @@ let decode_type_def v =
 	) in
 	(* if our package ends with an uppercase letter, then it's the module name *)
 	let pack,name = (match List.rev pack with
-		| last :: l when not (is_lower_ident last) -> List.rev l, last
+		| last :: l when (Lexer.is_valid_identifier last) && not (is_lower_ident last) -> List.rev l, last
```

Comment:
That means it now treats an upper-case invalid identifier as a part of the package, which seems weird. Something like `Type+` should error as a type, not as a package.

Reply:
Hm.. it's a bit tricky, without this, stuff like `""` and `0` just went through without any complaints.

Reply:
Maybe it just comes down to needing a check for module names as well.

Reply:
Oh, I thought you checked for that anyway.